source/common/http/http1/codec_impl.cc

```diff
@@ -259,7 +261,14 @@ const ToLowerTable& ConnectionImpl::toLowerTable() {
 }
 
 ConnectionImpl::ConnectionImpl(Network::Connection& connection, http_parser_type type)
-    : connection_(connection) {
+    : connection_(connection),
+      output_buffer_(Buffer::InstancePtr{new Buffer::OwnedImpl()},
+                     [&]() -> void { this->onOutputBufferBelowLowWatermark(); },
+                     [&]() -> void { this->onOutputBufferAboveHighWatermark(); }) {
+  uint32_t buffer_limit = connection.bufferLimit();
```

Comment:
nit: I feel like I've seen this code in different PRs. Since we are doing math here around / 2, etc., is it worth putting this into some small utility that takes a watermark buffer and a limit/connection?